native/components/message-markdown.react.js

```diff
@@ -0,0 +1,42 @@
+// @flow
+
+import * as React from 'react';
+import { View } from 'react-native';
+import * as SimpleMarkdown from 'simple-markdown';
+
+import { getMarkdownStyles } from '../themes/markdown-styles';
+import rules from './rules.react';
+
+type Props = {|
+  darkStyle: boolean,
+  children: string,
+|};
+function MessageMarkdown(props: Props) {
+  const style = React.useMemo(() => {
+    return getMarkdownStyles(props.darkStyle ? 'dark' : 'light');
+  }, [props.darkStyle]);
+
+  const customRules = React.useMemo(() => rules(style), [style]);
+
+  const parser = React.useMemo(() => SimpleMarkdown.parserFor(customRules), [
+    customRules,
+  ]);
+
+  const parse = React.useMemo(() => {
+    // we want to parse childen as one block, instead of inline
+    // to be interpreted as a one block, simple-markdown rules require \n\n at the end
+    // we're adding this manually, and specyfying inline as false, so block-parsing is forced
+    const blockSource = props.children + '\n\n';
+    return parser(blockSource, { inline: false });
```

Comment:
Reading through the code of `simple-markdown`, it looks like `inline: false` is the default, and if `disableAutoBlockNewlines: true` is not set, `simple-markdown` will append the newlines for us. So I think the `\n\n` and `{ inline: false }` are not necessary. We could simplify to:

```js
const parse = React.useMemo(
  () => parser(props.children),
  [parser, props.children],
);
```

That said, I think we might want to go with `{ inline: true }` – see my top-level comment about line breaks.

Reply:
If we go with `inline: true`, it will break in some cases, for example this: `[test](https://google.com) example` will be displayed like this
<img width="113" alt="Zrzut ekranu 2020-07-9 o 09 10 52" src="https://user-images.githubusercontent.com/18313471/87008525-29f69c80-c1c4-11ea-9081-ce08d0a93681.png">
instead of just 
<img width="124" alt="Zrzut ekranu 2020-07-9 o 09 11 47" src="https://user-images.githubusercontent.com/18313471/87008551-3b3fa900-c1c4-11ea-9090-75d689b21e1f.png">